### ISSUE TITLE
Inf China Minigunner in Helix has extended firing range when attacking air unit together with a grounded Minigunner

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#300]
Inf China Minigunner in Helix has extended firing range when attacking air unit together with a grounded Minigunner. Reported by MTKing.

> There's a similar situation that happens with inf helix, if garrisoned by a mini gunner and have the helix selected with another minigunner on the ground and shoot an air unit, the minigunner inside the lix will shoot the air unit with the same extended AA range

**Comment by commy2** [comment]
I don't really understand how this is a bug. The extended range is meant against air units. It's only a bug when used against ground units.

This is independent of whether the Minigunner is inside a Helix or not. I can also use the extended range against air without grouping the Assault Helix with Minigunner passenger with another Minigunner on ground.

**Comment by xezon** [comment]
I suspect what happens is, the Minigunner in the Helix will be able to acquire air target from further away with the assistance of the ground unit. Usually Minigunners in Helix will acquire target on rather short distance only.

**Comment by Stubbjax** [comment]
Minigunners inside a Helix will automatically acquire targets at a range much shorter than their actual attack range. Since it is not possible to target air units with a Helix, the Helix has to fly close to an air target before the Minigunners will begin firing. Selecting another unit that can target air in combination with the Helix allows players to attack air units without having to fly close to them first.

**Comment by MTKing4** [comment]
Yes, they get more range with this trick, without it they shoot the same way they shoot ground units meaning much shorter

**Comment by commy2** [comment]
Probably covered by my task here then: https://github.com/xezon/GeneralsGamePatch/blob/main/Patch104pZH/Design/Tasks/commy2_tasks.txt#L27

**Comment by MTKing4** [comment]
Should test the new changes then

**Comment by commy2** [comment]
Sure! It's not in yet though. And it's also tagged as MAYBE.

**Comment by MTKing4** [comment]
I will test the vanilla one then and show you how unfair it is

**Comment by MTKing4** [comment]
here it is


https://user-images.githubusercontent.com/83613147/133080647-8ec51221-df05-4887-812d-ed95c50ac24c.mp4

**Comment by xezon** [comment]
@commy2 
> Probably covered by my task here then:
> https://github.com/xezon/GeneralsGamePatch/blob/main/Patch104pZH/Design/Tasks/commy2_tasks.txt#L27

Firing range is quite extreme. Maybe we can decrease it for Minigunner when it sits in Helix.

@alanblack166 
> It may require a bunch of weapon refactors. If you activate `WeaponBonusPassedToPassengers` and give the object a `WeaponBonusUpgrade` module that is activated by default it will give them the `PLAYER_UPGRADE` weapon bonus flag. If the weapon has no existing `PLAYER_UPGRADE` bonuses, then you can use that to slightly modify the weapon via range, damage, rate of fire. But a complete weapon swap is impossible. Assuming China Infantry general has no infantry that use `WeaponBonusUpgrade` for upgrades, then weapon range adjustment in Helix is possible.

**Comment by Jundiyy** [comment]
> Firing range is quite extreme. Maybe we can decrease it for Minigunner when it sits in Helix.

It's a normal AA range with bonus, it's not bugged out or anything. (I'm referring to the Mini Gunner inside the Helix alone), it would still get this range without the bug, if it moves within range of an air unit then moves away, you gets this extreme range as the AA weapon is being used.

The bug itself, if this is the only fix for it then we prob don't have much other choices. Would result in changes to how a Helix is now micro'd vs air units, although it's not a bad change, just depends how players see it.

**Comment by Jundiyy** [comment]
> Firing range is quite extreme. Maybe we can decrease it for Minigunner when it sits in Helix.

Mini Gunner already has an extreme air range, placing it inside a vehicle makes it even better as you can move around and still have the range.

Regarding inside a Helix, it's a normal AA range with bonus, it's not bugged out or anything. (I'm referring to the Mini Gunner inside the Helix alone, not the bug), it would still get this range without the bug, if it moves within range of an air unit then moves away, you gets this extreme range as the AA weapon is being used.

The bug itself, if this is the only fix for it then we prob don't have much other choices. Would result in changes to how a Helix is now micro'd vs air units, although it's not a bad change, just depends how players see it. 

Edit: Just watched the change on the other page and Mini Gunner can shoot with AA weapon without coming in range first, so probs not the best solution.

**Comment by commy2** [comment]
@xezon If I understand this correctly, it would grant RPG-Soldiers (GLA) their AP Rockets when inside the Helix.

**Comment by Cryoblooded** [comment]
i think you don't know understand why they have this insane range it' becaus of the bunker upgrade gives extra range for evey inf unit inside the helix i think it should stay as it is
